### PR TITLE
Support repack feature #999

### LINKED ISSUE #999: Support repack feature
User can repack a bunch of objects based on the click order

image

### PR DESCRIPTION
**Pull request by Dawnlord**
Fixes #999

### CODE REVIEW

**Comment by damithc** [comment]
icon: may be make the objects different widths?

**Comment by Dawnlord** [comment]
qq 20160527094725

**Comment by damithc** [comment]
I think numbers are two small to read. No need for numbers. Colors show the order has been changed.

**Comment by Dawnlord** [comment]
qq 20160527095549

How about this one?

**Comment by damithc** [comment]
Icon is fine

**Comment by Dawnlord** [comment]
@kai33 @vhazali  who can help to review this?

File: PowerPointLabs/PowerPointLabs/PositionsLab/PositionsLabMain.cs [hunk 1]
```diff
@@ -2097,6 +2097,49 @@ public static void Swap(List<PPShape> selectedShapes, bool isPreview)
...
+                throw new Exception(ErrorMessageFewerThanTwoSelection);
+            }
+
+            var sortedShapes = selectedShapes;
+            System.Drawing.PointF[] topLeftPointList = new System.Drawing.PointF[numOfShapes];
+            sortedShapes = Graphics.SortShapesByLeft(selectedShapes);
+            float[] distanceList = new float[numOfShapes - 1];
+            var currentX = sortedShapes[0].ActualTopLeft.X;
+
+            for (var i = 0; i < numOfShapes - 1; i++)
```
Comment (fionachang) [inline PowerPointLabs/PowerPointLabs/PositionsLab/PositionsLabMain.cs]: Can you merge the first and second `for` loop together by initializing the `topLeftPointList[0]` then loop starting from 1 and use `i` and `i-1` instead? Can you also merge line 2109 and 2111?

**Comment by vhazali** [comment]
Don't think this is working as intended
Original layout:
image
The numbers in the box indicate the click order
Results after clicking on repack:
image

Based on what was previously discussed I think we want something like https://github.com/nus-fboa2016-PL/PowerPointLabs/issues/26 or in the example file: 
image

I think the whitespace between the shapes should be equalized

**Comment by Dawnlord** [comment]
Ok, I will check it. I think I may misunderstand this feature.

**Comment by IanTeo** [comment]
Closing this stale PR

**Pull request closed without merging**